docs/js/typescript.md

```diff
@@ -0,0 +1,6 @@
+# Typescript Guidelines
+
+- You MAY use Typescript to generate Javascript code.
+- Typescript files should be located in a subfolder `ts`. The generated Javascript files should be located in a subfolder `js` on the same level, e.g. `Services/Component/ts/component.ts` and `Services/Component/js/component.js`.
```

Comment:
Have you also considered `dist` or `build` as an  alternative? **If** yes, what was the deciding factor?

Reply:
Thanks for the feedback!

I have no preference and did not decide anything. :-) `dist` or `build` would be fine. An alterntaive for `ts` would be `src` in this case, right? What would this mean for all our other JS files that are mostly located in js subdirectories?

Reply:
Yes, `src` could/would be the alternative for `ts`.

The existing `./js` directories could be easily renamed, their paths used in PHP files could be changed by search&replace. But: Good point, all existing `js` folders are a valid argument to maintain this convention in future, even if this is uncommon for JS developers.

Reply:
Maybe we use another subdirectory structure for this and `./js` is our convetion for the parent. I will change the proposal accordingly. `build` may be more appropriate, since the final "distribution" would be not within the components directories at all (if bundled at some top level).